### ISSUE TITLE
TypeError: an integer is required  

### ISSUE THREAD

**Issue by tiru1930** [explosion/spaCy#2545]
I am trying to use spacy with some custom marcher functions, it had worked for few input files, but when i sent different input data it is giving below error.
an integer is required

```
 My reservation was for coach B but now that s 1st class so I m standing between carriages for 2 5hrs
<class 'str'>
Traceback (most recent call last):
  File "AIExpFrameWork.py", line 182, in <module>
    main()
  File "AIExpFrameWork.py", line 156, in main
    mp.readfile().clean().detectLanguage().anonymize().enhance().page_rank().encode().cluster().postprocess().nameIntents()
  File "/home/tiru/Desktop/cluster_pipline/make_pipeline.py", line 84, in clean
    self.raw_data['cleaned'] = self.raw_data['text'].apply(pipeline_utils.clean_simple_csv)
  File "/usr/local/lib/python3.5/dist-packages/pandas/core/series.py", line 3194, in apply
    mapped = lib.map_infer(values, f, convert=convert_dtype)
  File "pandas/_libs/src/inference.pyx", line 1472, in pandas._libs.lib.map_infer
  File "/home/tiru/Desktop/cluster_pipline/utils.py", line 28, in clean_simple_csv
    line = date_identification.main(matcher,text)
  File "/home/tiru/Desktop/cluster_pipline/date_identification.py", line 168, in main
    doc = nlp(text)
  File "/usr/local/lib/python3.5/dist-packages/spacy/language.py", line 346, in __call__
    doc = self.make_doc(text)
  File "/usr/local/lib/python3.5/dist-packages/spacy/language.py", line 378, in make_doc
    return self.tokenizer(text)
  File "tokenizer.pyx", line 116, in spacy.tokenizer.Tokenizer.__call__
  File "tokenizer.pyx", line 157, in spacy.tokenizer.Tokenizer._tokenize
  File "tokenizer.pyx", line 236, in spacy.tokenizer.Tokenizer._attach_tokens
  File "vocab.pyx", line 130, in spacy.vocab.Vocab.get
  File "vocab.pyx", line 166, in spacy.vocab.Vocab._new_lexeme
TypeError: an integer is required
```

## Info about spaCy

* **Models:** en_core_web_md, en
* **spaCy version:** 2.0.11
* **Python version:** 3.5.2
* **Platform:** Linux-4.13.0-45-generic-x86_64-with-Ubuntu-16.04-xenial

**Comment by ines** [comment]
Hi, could you post an isolated example of your code? The traceback you posted is really hard to follow.

In general, the `An integer is required` error means that you're passing in a string (e.g. for a label) instead of an integer (hash value of the string). You can usually solve that by looking up the string in the `StringStore`, for example:

```python
string_label = 'TEST'
integer_label = nlp.vocab.strings['TEST']
```

**Comment by tiru1930** [comment]
Hi 

 try:
        doc = nlp(text)
    except Exception as e:
        print(e)
        print(text)
        # print (ftfy.guess_bytes(text)[1])
        print(type(text))
        raise

when i am create nlp doc for each document in my file , it is working for  some documents, but doc like below which i am going to mention giving above error.

 My reservation was for coach B but now that s 1st class so I m standing between carriages for 2 5hrs

but add the same time when i create nlp doc for the above line on command line , it is working. Not able to figure it out what is the problem.

**Comment by ines** [comment]
This is strange – it looks like it fails on creating the lexical attributes, but I can't reproduce the error.

Can you share your `nlp` object? Which model did you load? Did you add any custom components?
